### PR TITLE
Misc docs updates

### PR DESCRIPTION
**Pull request by jeromelaban**
closes https://github.com/unoplatform/uno-private/issues/264

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]

**Review by dr1rrb — APPROVED** [review]

**Review by ajpinedam — APPROVED** [review]

File: doc/articles/features/working-with-xaml-hot-reload.md [hunk 1]
```diff
@@ -2,74 +2,91 @@
...
-
-The Uno Platform Hot Reload feature provides a way to modify the XAML displayed in a running application, in order to iterate faster on UI changes and Data Binding updates. This makes the inner developer loop faster.
+# Hot Reload
 
+The Uno Platform Hot Reload feature provides a way to modify the XAML and C# of your running application, in order to iterate faster on UI or code changes. This makes the inner developer loop faster.
 ## Features
-- XAML Hot Reload for iOS, Android, WebAssembly, Skia (Gtk, WebAssembly)
-- Supported in Visual Studio 2019, 2022 (Windows) and VS Code (Linux, macOS, Windows, CodeSpaces and GitPod)
+- XAML Hot Reload for iOS, Catalyst, Android, WebAssembly, Skia (Gtk and WPF)
+- Supported in Visual Studio 2022 (Windows) and VS Code (Linux, macOS, Windows, CodeSpaces and GitPod)
```
Comment (agneszitte) [inline doc/articles/features/working-with-xaml-hot-reload.md:10]:
```suggestion
- XAML Hot Reload for iOS, Catalyst, Android, WebAssembly, Skia (Gtk and WPF)
- Supported in Visual Studio 2022 (Windows) and VS Code (Linux, macOS, Windows, CodeSpaces, and GitPod)
```

File: doc/articles/features/working-with-xaml-hot-reload.md [hunk 2]
```diff
@@ -2,74 +2,91 @@
...
 - Supports XAML files in the main project, in shared projects, and referenced projects
-- Partial hot reload is supported, where modifying a UserControl instantiated in multiple locations will reload it without reloading its parents.
+- Partial tree hot reload is supported, where modifying a `UserControl`` instantiated in multiple locations will reload it without reloading its parents.
```
Comment (agneszitte) [inline doc/articles/features/working-with-xaml-hot-reload.md:12]:
```suggestion
- Partial tree hot reload is supported, where modifying a `UserControl` instantiated in multiple locations will reload it without reloading its parents.
```

File: doc/articles/features/working-with-xaml-hot-reload.md [hunk 3]
```diff
@@ -2,74 +2,91 @@
...
-    - Selecting the project name instead of IISExpress in the toolbar debug icon drop down list
+- The file named `RemoteControlGenerator\RemoteControl.g.cs` in the analyzers node for your project contains the connection information, verify that the information host addresses and the port number.
 - WinAppSDK on Windows specific issues
     - Grid Succinct syntax [is not supported](https://github.com/microsoft/microsoft-ui-xaml/issues/7043#issuecomment-1120061686)
 
-### Visual Studio 2019/2022
-- The output window in VS has an output named "Uno Platform" in its drop down. Diagnostics messages from the VS integration appear there.
-- Android
+### Visual Studio 2022
+- The output window in VS has an output named `Uno Platform` in its drop down. Diagnostics messages from the VS integration appear there.
```
Comment (agneszitte) [inline doc/articles/features/working-with-xaml-hot-reload.md:82]:
```suggestion
- The output window in VS has an output named `Uno Platform` in its drop-down. Diagnostics messages from the VS integration appear there.
```

File: doc/articles/migrating-from-previous-releases.md [hunk 4]
```diff
@@ -62,6 +62,9 @@ This behavior can be disabled by using `FeatureConfiguration.ResourceDictionary.
 #### `IsEnabled` property is moved from `FrameworkElement` to `Control`
 This property was incorrectly located on `FrameworkElement` but its behavior has not changed.
 
+#### `RegisterLoadActions` has been removed
+This method has been been removed as is it not available in WinUI. You can migrate code using this method to use the `FrameworkElement.Loaded` event instead.
```
Comment (agneszitte) [inline doc/articles/migrating-from-previous-releases.md:66]:
```suggestion
#### `RegisterLoadActions` has been removed
This method has been removed as is it not available in WinUI. You can migrate code using this method to use the `FrameworkElement.Loaded` event instead.
```

File: doc/articles/migrating-from-previous-releases.md [hunk 5]
```diff
@@ -62,6 +62,9 @@ This behavior can be disabled by using `FeatureConfiguration.ResourceDictionary.
 #### `IsEnabled` property is moved from `FrameworkElement` to `Control`
 This property was incorrectly located on `FrameworkElement` but its behavior has not changed.
 
+#### `RegisterLoadActions` has been removed
+This method has been removed as is it not available in WinUI. You can migrate code using this method to use the `FrameworkElement.Loaded` event instead.
```
Comment (Xiaoy312) [inline doc/articles/migrating-from-previous-releases.md:66]:
```suggestion
This method has been removed as it is not available in WinUI. You can migrate code using this method to use the `FrameworkElement.Loaded` event instead.
```

File: doc/articles/features/working-with-xaml-hot-reload.md [hunk 6]
```diff
@@ -2,74 +2,91 @@
...
+- In Visual Studio for Windows:
+  - If your app is using port 5000 in development, [this Visual Studio issue](https://developercommunity.visualstudio.com/t/BrowserLink-WebSocket-is-disconnecting-a/10500228) may cause problems, changing to another port in `Properties/launchSettings.json` (e.g. 5001) will work around it.
+  - [`MetadataUpdateHandlers`](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.metadata.metadataupdatehandlerattribute?view=net-7.0) are invoked without the list of changed types, which means that some hot reload features may not be available.
 
-### Visual Studio Code
-- Follow the [getting started guide](../get-started-vscode.md)
-- Start the application (WebAssembly or Skia)
-- Make modifications to XAML files and save the file
+> [!IMPORTANT]
+> If your solution contains iOS/Android/Catalyst or WebAssembly project heads, Hot Reload will only be available when targeting .NET 8 or later (`net8.0` in the `TargetFrameworks` property)
```
Comment (Xiaoy312) [inline doc/articles/features/working-with-xaml-hot-reload.md:57]: should this block be moved out of `#tab/wasm`? or duplicated under `#tab/mobiles`?

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

**Pull request merged**